packages/reactivity/src/effect.ts

```diff
@@ -341,19 +352,19 @@ export function trigger(
       }
     }
     if (__DEV__) {
-      triggerEffects(createDep(effects), eventInfo)
+      triggerEffects(new Set(effects), eventInfo)
     } else {
-      triggerEffects(createDep(effects))
+      triggerEffects(new Set(effects))
     }
   }
 }
 
 export function triggerEffects(
-  dep: Dep | ReactiveEffect[],
+  dep: Set<ReactiveEffect>,
   debuggerEventExtraInfo?: DebuggerEventExtraInfo
 ) {
   // spread into array for stabilization
-  const effects = isArray(dep) ? dep : [...dep]
+  const effects = [...dep]
```

Comment:
These changes were inspired by #6185.

The `createDep(effects)` call isn't really creating a `Dep`, it's just using a `Set` to remove duplicates. This is the only place that passes an argument to `createDep()`, and I wanted to change the signature of `createDep()` to be able to pass the `target` and `key` instead. The other changes here follow from that.

Reply:
I think you can keep the signature of triggerEffects, then create the Set inside the triggerEffects if needed.